catalog/spatial/geocoding/spatial-geocoding-feature/src/main/java/org/codice/ddf/spatial/geocoding/feature/CatalogFeatureQueryable.java

```diff
@@ -0,0 +1,100 @@
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.geocoding.feature;
+
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.io.ParseException;
+import com.vividsolutions.jts.io.WKTReader;
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.SourceResponse;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.apache.commons.lang.Validate;
+import org.codice.ddf.spatial.geocoding.FeatureQueryException;
+import org.codice.ddf.spatial.geocoding.FeatureQueryable;
+import org.geotools.feature.simple.SimpleFeatureBuilder;
+import org.opengis.feature.simple.SimpleFeature;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CatalogFeatureQueryable implements FeatureQueryable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CatalogFeatureQueryable.class);
+
+  private static final ThreadLocal<WKTReader> WKT_READER_THREAD_LOCAL =
+      ThreadLocal.withInitial(WKTReader::new);
+
+  private CatalogFramework catalogFramework;
+
+  private CatalogHelper catalogHelper;
+
+  public CatalogFeatureQueryable(CatalogFramework catalogFramework, CatalogHelper catalogHelper) {
+    this.catalogFramework = catalogFramework;
+    this.catalogHelper = catalogHelper;
+  }
+
+  @Override
+  public List<SimpleFeature> query(String queryString, String featureCode, int maxResults)
+      throws FeatureQueryException {
+    Validate.notNull(queryString, "queryString can't be null");
+
+    if (maxResults < 0) {
+      throw new IllegalArgumentException("maxResults can't be negative");
+    }
+
+    Query query = catalogHelper.getQueryForCountryCode(queryString);
+    QueryRequest queryRequest = new QueryRequestImpl(query);
+
+    SourceResponse response;
+    try {
+      response = catalogFramework.query(queryRequest);
+    } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+      LOGGER.debug("Failed to query catalog for feature {}", queryString, e);
```

Comment:
❗️ log and throw code smell (http://rolf-engelhard.de/2013/04/logging-anti-patterns-part-ii/)

Reply:
removed logging